File: 383.Ransom_Note/solution.py
def canConstruct(ransomNote=str, magazine=str) -> bool:
    myCheckingString = ""
    for i in ransomNote:
        if i in magazine:
            magazine = magazine.replace(i, "", 1)
            myCheckingString += i
    return True if myCheckingString == ransomNote else False


# case 1
print(canConstruct("a", "b"))  # False

# case 2
print(canConstruct("aa", "ab"))  # False

# case 3
print(canConstruct("aa", "aab"))  # True

File: 383.Ransom_Note/solution.js
// * Accepted

/**
 * @param {string} ransomNote
 * @param {string} magazine
 * @return {boolean}
 */
var canConstruct = function (ransomNote, magazine) {
  for (let i = 0; i < ransomNote.length; i++) {
    const el = ransomNote[i];
    if (magazine.indexOf(el) != -1) {
      magazine = magazine.replace(el, "");
      // // continue;
    } else {
      return false;
    }
  }
  return true;
};

// Case 1
console.log(canConstruct("a", "b")); // ? false

// Case 2
console.log(canConstruct("aa", "ab")); // ? false

// Case 3
console.log(canConstruct("aa", "aab")); // ? true
